### PR TITLE
Sqlas read ir save pabaigtas

### PR DESCRIPTION
**Pull request by Justuxs**

### CODE REVIEW

File: CityPuzzle/Classes/Room.cs [hunk 1]
```diff
@@ -9,16 +9,12 @@ namespace CityPuzzle.Classes
     public class Room
     {
         public String ID { get; set; }
-        public String Owner { get; set; }
+        public int Owner { get; set; }
```
Comment (Justuxs) [inline CityPuzzle/Classes/Room.cs:12]: Owneri saugau ne kaip username bet kaip ID- del galimo ateityje lenteliu jungimo, o ID yra primary key

File: CityPuzzle/Classes/Sql.cs [hunk 2]
```diff
@@ -27,7 +30,7 @@ public static void SaveUser(User user)
             }
         }
 
-        public static String ReadUsers()
+        public static List<User> ReadUsers()
```
Comment (Justuxs) [inline CityPuzzle/Classes/Sql.cs:33]: Pataisiau, nes buvo neasmone grazinama

File: CityPuzzle/Classes/Sql.cs [hunk 3]
```diff
@@ -68,16 +67,50 @@ public static void SavePuzzle(Puzzle puzzle)
...
                 command.Parameters.AddWithValue("@About", puzzle.About);
                 command.Parameters.AddWithValue("@Quest", puzzle.Quest);
                 command.Parameters.AddWithValue("@Latitude", puzzle.Latitude);
                 command.Parameters.AddWithValue("@Longitude", puzzle.Longitude);
+                command.Parameters.AddWithValue("@ImgAdress", puzzle.ImgAdress);
                 command.ExecuteNonQuery();
                 conn.Close();
             }
         }
+        public static String ReadPuzzles()
```
Comment (AllBill127) [inline CityPuzzle/Classes/Sql.cs]: Shouldn't all read methods return list of items?

File: CityPuzzle/Classes/Sql.cs [hunk 4]
```diff
@@ -68,16 +67,50 @@ public static void SavePuzzle(Puzzle puzzle)
...
+                String sql, Output = "";
 
+                sql = "Select ID,Name,About,Quest,Latitude,Longitude,ImgAdress from Puzzles";
+                conn.Open();
+                command = new SqlCommand(sql, conn);
+                dataReader = command.ExecuteReader();
+                List<Puzzle> puzzles = new List<Puzzle>();
+                while (dataReader.Read())
+                {
+                    Output = Output + dataReader.GetValue(0) + "-" + dataReader.GetValue(1) + "-" + dataReader.GetValue(3) + "-" + dataReader.GetValue(4) + "-" + dataReader.GetValue(5);
```
Comment (AllBill127) [inline CityPuzzle/Classes/Sql.cs:98]: Output strings are not necessary

File: CityPuzzle/Classes/Room.cs [hunk 5]
```diff
@@ -9,16 +9,12 @@ namespace CityPuzzle.Classes
...
         public int RoomSize { get; set; }
         [TextBlob("addressesBlobbed")]
```
Comment (AllBill127) [inline CityPuzzle/Classes/Room.cs:14]: Delete unused line

File: CityPuzzle/Classes/Sql.cs [hunk 6]
```diff
@@ -112,5 +113,23 @@ public static String ReadPuzzles()
...
         }
+        public static void SaveRoom(Room room)
+        {
+            // Func<List <Puzzle>, string> convert =
+            var numbers = room.Tasks.Select(x => x.ID);
+            string converted=string.Join("", numbers+"-");
+            using (SqlConnection conn = new SqlConnection(ConnStr))
+            {
+                conn.Open();
+                var command = new SqlCommand("INSERT INTO Rooms (ID,Owner,RoomSize,Tasks) VALUES (@ID,@Owner,@RoomSize,@Tasks)", conn);
```
Comment (AllBill127) [inline CityPuzzle/Classes/Sql.cs:125]: Mismatch on command parametres given and parametres being set. Tasks and Latitude.
Also missing Room Participants

File: CityPuzzle/Classes/Sql.cs (hunk 2 above)
Comment (AllBill127) [reply CityPuzzle/Classes/Sql.cs:33]: good idea

File: CityPuzzle/Classes/Sql.cs (hunk 3 above)
Comment (AllBill127) [reply CityPuzzle/Classes/Sql.cs]: got fixed in next commit. very good

File: CityPuzzle/Classes/Sql.cs [hunk 7]
```diff
@@ -110,15 +160,19 @@ public static void SavePuzzle(Puzzle puzzle)
...
             }
         }
         public static void SaveRoom(Room room)
         {
             // Func<List <Puzzle>, string> convert =
-            var numbers = room.Tasks.Select(x => x.ID);
-            string converted=string.Join("", numbers+"-");
+            var numbers = room.Tasks.Select(x => x.Value.ID);
+            string converted = string.Join("", numbers + "-");
             using (SqlConnection conn = new SqlConnection(ConnStr))
```
Comment (AllBill127) [inline CityPuzzle/Classes/Sql.cs:176]: 179 line (@Tasks) doesnt match 183 line (@Latitude). What is happening there?

**Comment by AllBill127** [comment]
Resolved merge conflicts from older version of sqlas branch

**Pull request merged**